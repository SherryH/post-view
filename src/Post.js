import React from 'react';
import styles from './post.css';

const Post = ({userId, title, body}) => (
  <div className={styles.post}>
    <div className={styles.postUser}>{userId}</div>
    <div>
      <div className={styles.postTitle}>{title}</div>
      <div>{body}</div>
    </div>
  </div>
);

export default Post;