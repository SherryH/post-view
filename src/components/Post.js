import React from 'react';
import styles from './post.css';

const Post = ({ userId, title, body }) => (
  <div className={styles.post}>
    <div className={styles.postUser}>{userId}</div>
    <div>
      <div className={styles.postTitle}>{title}</div>
      <div>{body}</div>
    </div>
  </div>
);

Post.defaultProps = {
  userId: '',
  title: '',
  body: '',
};

Post.propTypes = {
  userId: React.PropTypes.number,
  title: React.PropTypes.string,
  body: React.PropTypes.string,
};

export default Post;
