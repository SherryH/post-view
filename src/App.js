import React from 'react';
import Post from './Post';
import styles from './app.css';

const dummyData = [
  {
    "userId": 1,
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
  },
  {
    "userId": 1,
    "id": 2,
    "title": "qui est esse",
    "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
  }];

const App = () => {
  return (
    <div id={styles.postList}>
      A list of posts
      {
        dummyData.map((data) => (
          <Post key={data.id.toString()} userId={data.userId} title={data.title} body={data.body}/>
        ))
      }
    </div>
  );
};

export default App;