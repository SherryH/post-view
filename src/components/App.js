import React from 'react';
import Post from './Post';
import styles from './app.css';

function App(props) {
  return (
    <div id={styles.postList}>
      <h2>A list of posts </h2>
      {
        props.currentPosts.map(post => (
          <Post key={post.id.toString()} userId={post.userId} title={post.title} body={post.body} />
        ))
      }
      <button onClick={props.showMorePosts}>Show More</button>
    </div>
  );
}

App.defaultProps = {
  currentPosts: [],
  showMorePosts: function showMorePosts() {},
};

App.propTypes = {
  showMorePosts: React.PropTypes.func,
  currentPosts: React.PropTypes.arrayOf(
    React.PropTypes.shape({
      body: React.PropTypes.string,
      id: React.PropTypes.number,
      title: React.PropTypes.string,
      userId: React.PropTypes.number,
    }),
  ),
};

export default App;
