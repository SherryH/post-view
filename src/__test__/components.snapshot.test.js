import React from 'react';
import renderer from 'react-test-renderer';
import Post from '../components/Post';
import App from '../components/App';

describe ('Component presentation:', () => {

  it ('Post component renders correctly', () => {
    const post = { userId: 1, title: 'dummyTitle', body: 'dummyBody' };
    const rendered = renderer.create(
      <Post post={post} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it ('App component renders correctly', () => {
    const currentPosts = [{
      id: 2,
      userId: 2,
      title: 'title2',
      body: 'body2'
    },{
      id: 3,
      userId: 3,
      title: 'title3',
      body: 'body3'
    }];
    const rendered = renderer.create(
      <App currentPosts={currentPosts} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
