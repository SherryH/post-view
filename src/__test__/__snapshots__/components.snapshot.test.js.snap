// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component presentation: App component renders correctly 1`] = `
<div
  id="postList"
>
  <h2>
    A list of posts 
  </h2>
  <div
    className="post"
  >
    <div
      className="postUser"
    >
      2
    </div>
    <div>
      <div
        className="postTitle"
      >
        title2
      </div>
      <div>
        body2
      </div>
    </div>
  </div>
  <div
    className="post"
  >
    <div
      className="postUser"
    >
      3
    </div>
    <div>
      <div
        className="postTitle"
      >
        title3
      </div>
      <div>
        body3
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Show More
  </button>
</div>
`;

exports[`Component presentation: Post component renders correctly 1`] = `
<div
  className="post"
>
  <div
    className="postUser"
  >
    1
  </div>
  <div>
    <div
      className="postTitle"
    >
      dummyTitle
    </div>
    <div>
      dummyBody
    </div>
  </div>
</div>
`;
